### ISSUE TITLE
Make bitcoin-s blockfilter & bitcoind backends use the same set of callbacks

### ISSUE THREAD

**Issue by Christewart** [bitcoin-s/bitcoin-s#3283]
> > So i think the better question to ask here is:
> > Is there any time we want to call `wallet.processBlock()` and NOT call `updateUtxoPendingStates()`?
> > If not, we should just move this method call into `wallet.processBlock()` and remove other places we have to call it manually. Thoughts?
> 
> We only call process block on blocks that we have a filter match, this means that we wouldn't update the utxo state until we spent or received another tx. Doing it this way means we just need to actually wait for the confirmations

So the reason love this doesn't get executed is because we never have `NodeCallbacks` added to bitcoind? 

https://github.com/bitcoin-s/bitcoin-s/blob/63083bf6425f0854053a779ce407702247b32435/app/server/src/main/scala/org/bitcoins/server/BitcoinSServerMain.scala#L220

We may want to think of a way to re-use the callbacks stuff between `BitcoindRpcAppConfig` and `NodeAppConfig` in the future so that we don't end up having this divergence between two bitcoin-s backends that retrieve information from the network. I don't think we necessarily want to do this on this PR, but it seems like something we should work towards in the future.

_Originally posted by @Christewart in https://github.com/bitcoin-s/bitcoin-s/issues/3276#issuecomment-862510089_

TLDR, we should have functionally equivalent callbacks registered regardless if blockfilters or bitcoind is being used as a backend
